---
title: "miRNA expression"
author: "SÃ©bastien Renaut (sebastien.renaut@criucpq.ulaval.ca)"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: F
params:
  resultspath: 'C:/Users/renseb01/Documents/Steinberg_Christian/miRNA/results/'
  figurespath: 'C:/Users/renseb01/Documents/Steinberg_Christian/miRNA/figures'
---



```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = params$outputpath)
knitr::opts_chunk$set(echo = F)
library(ggplot2)
library(patchwork)
library(DESeq2)
library(dplyr)
library(tidyr)
library(RColorBrewer)
library(ggrepel)
```


# Trimming & QC
  * Total reads processed:              10,446,968
  * Reads with adapters:                10,382,220 (99.4%)
  * Reads removed (<16bp):              1,158,793
  * Reads written (passing filters):    9,288,175 (88.9%)

  * Total base-pairs processed:         1,055,143,768 bp
  * Quality-trimmed:                    95,440,235 bp (9.0%)
  * Total written (filtered):           233,886,560 bp (22.2%)
  
  
  
#check what? 
```{r data}
dataset = c('bams','bams_REFRAME_2025')
d=2

files = list.files(file.path(params$resultspath,dataset[d]),recursive = TRUE,pattern = 'idxstats$')

for(i in 1:length(files)){
  
  temp = read.table(file.path(params$resultspath,dataset[d],files[i]),header =F)
  temp = temp[-nrow(temp),]
  if(i == 1) mirna_expression1 = data.frame(expression = temp$V3)
  
  if(i >1) mirna_expression1 = cbind(mirna_expression1,temp$V3)
      
  if(i == length(files)) {
    lengths = temp$V2
    names = sapply(strsplit(files,'_',fixed = T),"[",4);
    colnames(mirna_expression1) = gsub('.idxstats','',names);
    rownames(mirna_expression1) = temp$V1}
} 

#
rowm = rowMeans(mirna_expression1[,1:(ncol(mirna_expression1)-1)])
mirna_expression = mirna_expression1[order(rowm,decreasing = T),]
mirna_expression$mirna = rownames(mirna_expression)
mirna_expression$mirna = factor(mirna_expression$mirna,levels = mirna_expression$mirna)

mirna_expression_pivot = pivot_longer(mirna_expression,cols = 1:(ncol(mirna_expression1)-1),values_to = 'expression')
mirna_expression_pivot$type = ifelse(regexpr('CASE',mirna_expression_pivot$name)>0,'CASE','CONTROL')
top10 = head(mirna_expression$mirna,10)

rowm =rowMeans(mirna_expression[,1:12])

df = data.frame(`Number of miRNA`= c(nrow(mirna_expression),
           nrow(mirna_expression[rowm<1,]),                          
           nrow(mirna_expression[rowm>1,]),
           nrow(mirna_expression[rowm>5,]),
           nrow(mirna_expression[rowm>10,]),
           nrow(mirna_expression[rowm>100,]),
           nrow(mirna_expression[rowm>1000,]),
           nrow(mirna_expression[rowm>10000,]),
           nrow(mirna_expression[rowm>100000,]),
           nrow(mirna_expression[rowm>1000000,])),
           `Number or reads aligned per sample`  = c('Total','<1','>1','>5','>10','>100','>1k','>10k','>100k','>1M'),
           check.names = F
           )

df$`Number or reads aligned per sample` = factor(df$`Number or reads aligned per sample`, levels = c('Total','<1','>1','>5','>10','>100','>1k','>10k','>100k','>1M'))
```


```{r plots}
#histogram
histograms = mirna_expression_pivot %>% 
    ggplot(aes(x = log(expression,10),fill = type)) + 
    geom_histogram(alpha=0.5, position="identity") +
    theme_bw() +
    xlab(bquote(log[10]~Number~of~reads~aligned)) +
    theme(legend.position= 'inside',legend.position.inside = c(0.8, 0.8),
                legend.background = element_rect(fill="lightgray")) +
    ggtitle('Number of aligned sequences per miRNA')
  
 
#Boxplot
boxplot = mirna_expression_pivot %>%
  filter(mirna %in% top10) %>%
  ggplot(aes(x=mirna, y = expression,fill = type)) + 
  geom_boxplot(outliers = F,alpha = 0.5) +
  ylab('Number of reads') +
  xlab('miRNAs') +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1),
       legend.position= 'inside',legend.position.inside = c(0.8, 0.8),
                legend.background = element_rect(fill="lightgray")) +
  ggtitle('Top 10 most expressed miRNA') 

#Totals
totals = ggplot(df, aes(x = `Number or reads aligned per sample`, y = `Number of miRNA`)) +
    geom_bar(stat="identity", fill="steelblue")+
    geom_text(aes(label=`Number of miRNA`), vjust=c(rep(1.6,5),rep(-0.5,5)), color=c(rep("white",5),rep('black',5)), size=3.5)+
    theme_bw() +
    ggtitle('number of miRNAs')
```


```{r degs}
conditions = ifelse(regexpr('CASE',colnames(mirna_expression)[-13])>0,'CASE','CONTROL') 

dds <- DESeqDataSetFromMatrix(mirna_expression[,-ncol(mirna_expression)], DataFrame(conditions), ~ conditions)
     dds = DESeq(dds)
     resultsNames(dds)
     res = results(dds)
     res = res[order(res$pvalue),] #order by pval
     res = res[res$baseMean>0,] #remove 0 expression values
     res = res[!is.na(res$padj),] #remove NA


#
res$miRNA  = rownames(res)     
res$minuslogpadj = -log(res$padj,10)
res$genes = 'non-significant'
res$genes[res$minuslogpadj>5.1 | abs(res$log2FoldChange)>1.6] = 'significant'

annot = res[res$genes == 'significant', ]
#annot = annot[1:5,]
#
volcano_plot = ggplot(res,aes(x = log2FoldChange, y = minuslogpadj,color = genes)) + 
  geom_point() +
  xlim(c(-4,4)) +
  ylim(c(-1,20)) +
  geom_text_repel(data = annot,aes(x = log2FoldChange, y = minuslogpadj,label = miRNA),color = 'black') +
  ggtitle('Differential Expression (Case vs. Control)') +
  ylab( bquote(-log[10]~(p[adjusted]))) +
  xlab( bquote(log[2]~Fold~change)) +
  scale_color_discrete(type  = brewer.pal(9,'Set1')[c(1,2)]) +
  theme_bw() +
  theme(legend.position= 'none',legend.position.inside = c(0.8, 0.8),
                legend.background = element_rect(fill="lightgray"))
```

#PCA
```{r PCA, message= T}
library(FactoMineR)
#library(factoextra)
type = ifelse(regexpr('CASE',colnames(mirna_expression)[1:(ncol(mirna_expression)-1)])>0,'CASE','CONTROL')


res.pca = PCA(t(mirna_expression[,1:(ncol(mirna_expression)-1)]),scale = T, graph = F,ncp = 50)

fviz_pca_ind(res.pca, col.ind = type,repel = TRUE, geom.ind = 'point')


```



# save it
```{r session, message= T}
#save it 
pdf(file.path(params$figurespath,paste0('Figure1_miRNA',dataset[d],'.pdf')),width = 12,height = 8)
totals + histograms + boxplot + volcano_plot + plot_annotation(tag_levels = 'A')
dev.off()
```


# session info 
```{r session, message= T}

sessionInfo()

```
     
     
